File: api/src/main/java/com/example/api/entities/Article.java
package com.example.api.entities;





import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.io.Serializable;

@Entity
@Data @AllArgsConstructor
@NoArgsConstructor
@ToString
@Table(name="articles")
public class Article implements Serializable {
    @Id @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;
    private String name;
    private double price;
    private String picture;
    @ManyToOne
    private Order order;

}

File: client/src/app/models/Article.ts
import {Order} from "./Order";

export class Article {
  id?:number;
  name?:string;
  price?:boolean;
  picture?:string;



  // @ts-ignore
  constructor() {

  }
  // @ts-ignore
  constructor(data: any) {
    this.id = data.id;
    this.name = data.name;
    this.price = data.price;
    this.picture = data.picture;
  }











}
